File: agent-api-server/app/models/agent.py
from sqlalchemy import Column, String, Text, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from .base import Base
import uuid


class AIAgent(Base):
    __tablename__ = "ai_agents"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    name = Column(String(255), nullable=False)
    description = Column(Text)
    workflow_id = Column(String(36), ForeignKey("workflows.id"), nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    # Relationships
    workflow = relationship("Workflow", back_populates="agents")
    conversations = relationship("Conversation", back_populates="agent")

File: src/types/agent.ts
import { BaseEntity } from './common';

// AI Agent types
export interface AIAgent extends BaseEntity {
  name: string;
  description: string;
  enabled: boolean;
  preview_enabled: boolean;
  workflow_id?: string;
}

export interface CreateAgentRequest {
  name: string;
  description: string;
  enabled: boolean;
  preview_enabled: boolean;
}

export interface UpdateAgentRequest {
  name?: string;
  description?: string;
  enabled?: boolean;
  preview_enabled?: boolean;
  workflow_id?: string;
}
